### PR TITLE
[OSDOCS-10179]:Upgrading machine pools in ROSA

### PR DESCRIPTION
**Pull request by mletalie**
Version(s):

4.15+
Issue:

https://issues.redhat.com/browse/OSDOCS-10179
Link to docs preview:

https://74750--ocpdocs-pr.netlify.app/openshift-rosa/latest/cli_reference/rosa_cli/rosa-manage-objects-cli.html#rosa-upgrade-machinepool_rosa-managing-objects-cli
QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Comment by mletalie** [comment]
Per https://redhat-internal.slack.com/archives/C0308P3FC7R/p1713314177151809?thread_ts=1713303198.310109&cid=C0308P3FC7R, can you please QE review, @yingzhanredhat? Thanks.

**Comment by bchandra-ocp** [comment]
1. Delete upgrade of machine pool is missing. $ rosa delete upgrade --cluster {} --machinepool {}

**Review by bchandra-ocp — CHANGES_REQUESTED** [review]
Similar to deleting scheduled upgrades of cluster, we need to document deletion of machine pool upgrades.

File: modules/rosa-upgrade-cluster-cli.adoc [hunk 1]
```diff
@@ -101,3 +101,95 @@ $ rosa delete upgrade --cluster=<cluster_name> | <cluster_id>
...
+
+|--profile
+|Specifies an AWS profile (string) from your credentials file.
+|===
+
+.Example
+Upgrade a machine pool named `mymachinepool` on a cluster named `mycluster`.
+[source,terminal]
+----
+$ rosa upgrade machinepool --cluster=mycluster --machinepool=mymachinepool
```
Comment (yingzhanredhat) [inline modules/rosa-upgrade-cluster-cli.adoc:148]: It is not correct. It should like this
`rosa upgrade machinepool --cluster=mycluster --machinepool=mymachinepool`

File: modules/rosa-upgrade-cluster-cli.adoc [hunk 2]
```diff
@@ -101,3 +101,95 @@ $ rosa delete upgrade --cluster=<cluster_name> | <cluster_id>
...
+
+[id="rosa-delete-upgrade-machinepool_{context}"]
+== delete machinepool upgrade
+
+Cancel a scheduled machinepool upgrade.
+
+.Syntax
+[source,terminal]
+----
+$ rosa delete upgrade --cluster=<cluster_name> --machinepool=<machinepool_name>| <cluster_id> <machinepool_id> [arguments]
```
Comment (yingzhanredhat) [inline modules/rosa-upgrade-cluster-cli.adoc:159]: No "--machinepool" flag

**Comment by mletalie** [comment]
@yingzhanredhat, 

    I have updated docs based on your comment. Please let me know if that note specifying that the upgraded command is for hcp clusters only, or if it should be just applied to upgrading machine pools. Once that is resolved, I believe the docs will be ready to be published. Thanks.

**Comment by mletalie** [comment]
> @mletalie Sure.
> 
> 1. upgrade machinepool is only supported for HCP cluster
> 2. upgrade cluster is supported for classic and HCP.
>    For upgrade cluster,
> 
> * --node-drain-grace-period is only supported for classic cluster
> * --control-plane is only supported for HCP cluster

Hello @yingzhanredhat, thanks for this info. I see that `--control plane ` flag is not listed at all currently in the docs, and is outside the scope of this ticket. That being said, I can add it to the docs as part of this doc, but would need to know what the flag does? I can see about checking myself but given this ticket is getting outside of scope, if you could let me know it would save some time. Thanks!

**Comment by eohartman** [comment]
Looks great!

**Comment by stevsmit** [comment]
lgtm

**Pull request merged**